src/v2/Apps/Settings/Routes/SavedSearchAlerts/Components/__tests__/SavedSearchAlertEditForm.jest.tsx

```diff
@@ -0,0 +1,245 @@
+      target: { value: "Updated Name" },
+    })
+
+    const saveAlertButton = screen.getByRole("button", {
+      name: "Save Alert",
+    })
+
+    fireEvent.click(saveAlertButton)
+
+    await waitFor(() => expect(mockOnCompleted).toBeCalled())
+  })
+
+  describe("Save Alert button", () => {
+    it("should be disabled by default", () => {
+      renderWithRelay({
+        Artist: () => artistMocked,
+        FilterArtworksConnection: () => filterArtworksConnectionMocked,
+        SearchCriteria: () => savedSearchAlertMocked,
+      })
+      const saveAlertButton = screen.getByRole("button", {
+        name: "Save Alert",
+      })
+
+      expect(saveAlertButton).toBeDisabled()
+    })
+
+    it("should be enabled if alert name is changed", () => {
+      renderWithRelay({
+        Artist: () => artistMocked,
+        FilterArtworksConnection: () => filterArtworksConnectionMocked,
+        SearchCriteria: () => savedSearchAlertMocked,
+      })
+
+      fireEvent.change(screen.getByDisplayValue("Alert #1"), {
+        target: { value: "Updated Name" },
+      })
+
+      const saveAlertButton = screen.getByRole("button", {
+        name: "Save Alert",
+      })
+
+      expect(saveAlertButton).toBeEnabled()
+    })
+
+    it("should be disabled if all notification checkboxes are disabled", () => {
+      renderWithRelay({
+        Artist: () => artistMocked,
+        FilterArtworksConnection: () => filterArtworksConnectionMocked,
+        SearchCriteria: () => savedSearchAlertMocked,
+      })
+      const checkbox = screen.getByRole("checkbox", {
+        name: "Email Alerts",
+      })
+
+      fireEvent.click(checkbox)
+
+      const saveAlertButton = screen.getByRole("button", {
+        name: "Save Alert",
+      })
+
+      expect(saveAlertButton).toBeDisabled()
+    })
+
+    it("should be enabled if pills are changed", () => {
+      renderWithRelay({
+        Artist: () => artistMocked,
+        FilterArtworksConnection: () => filterArtworksConnectionMocked,
+        SearchCriteria: () => savedSearchAlertMocked,
+      })
+
+      fireEvent.click(screen.getByText("Buy Now"))
+
+      const saveAlertButton = screen.getByRole("button", {
+        name: "Save Alert",
+      })
+
+      expect(saveAlertButton).toBeEnabled()
+    })
+
+    it("should be enabled if notification checkboxes are changed", () => {
+      renderWithRelay({
+        Artist: () => artistMocked,
+        FilterArtworksConnection: () => filterArtworksConnectionMocked,
+        SearchCriteria: () => savedSearchAlertMocked,
+      })
+      const checkbox = screen.getByRole("checkbox", {
+        name: "Mobile Alerts",
+      })
+
+      fireEvent.click(checkbox)
+
+      const saveAlertButton = screen.getByRole("button", {
+        name: "Save Alert",
+      })
+
+      expect(saveAlertButton).toBeEnabled()
+    })
+  })
+})
+
+const artistMocked = {
+  internalID: "artistId",
+  name: "Banksy",
+  slug: "banksy",
+}
+
+const savedSearchAlertMocked = {
```

Comment:
**question**: Would it be helpful to move these mocks to a fixtures file?

Reply:
I think that for now we can leave it here and in the future (if we need to use it for other tests), then move to `fixtures` folder